pgdrive/utils/scene_utils.py

```diff
@@ -132,3 +118,28 @@ def get_all_lanes(roadnet: "RoadNetwork"):
             for l in lanes:
                 res.append(l)
     return res
+
+
+def ray_localization(position: np.ndarray, pg_world: PgWorld) -> Tuple:
+    """
+    Get the index of the lane closest to a physx_world position.
+    Only used when smoething is on lane ! Otherwise fall back to use get_closest_lane()
+    :param position: a physx_world position [m].
+    :param pg_world: PgWorld class
+    :return: the index of the closest lane.
+    """
+    results = pg_world.physics_world.static_world.rayTestAll(
+        Vec3(position[0], -position[1], 1.0), Vec3(position[0], -position[1], -1)
```

Comment:
Why a `-position[1]` here? Where does the negative comes from? I think we should stick to a coordinate system all the time.

Reply:
```rayTest()``` is API of panda3d, and the ```position``` here is in PGDrive's coordinate. I will add some methods in ```basic_utils.py``` to do these transform like ```heading```, ```position```, ```direction```.